packages/modular-scripts/src/cli.ts

```diff
@@ -118,6 +128,38 @@ function run() {
   }
 }
 
+interface WorkspaceMeta {
+  location: string;
+  workspaceDependencies: string[];
+  mismatchedWorkspaceDependencies: string[];
+}
+
+function listPackages(typeArg: string | void) {
+  const modularRoot = getModularRoot();
+
+  const { stdout } = execa.sync('yarnpkg', ['workspaces', 'info'], {
+    cwd: modularRoot,
+  });
+  const workspaceInfo: Record<string, WorkspaceMeta> = JSON.parse(
+    stdout,
+  ) as Record<string, WorkspaceMeta>;
+
+  const packages = Object.values(workspaceInfo)
+    .filter((workspaceMeta: WorkspaceMeta) => {
+      const location = path.join(modularRoot, workspaceMeta.location);
+      if (typeArg) {
+        return isModularType(location, typeArg as PackageType);
+      } else {
+        return isModularPackage(location);
+      }
+    })
+    .map((workspaceMeta: WorkspaceMeta) => {
+      return path.basename(workspaceMeta.location);
+    });
```

Comment:
For consistency with the rest of the codebase where we use implicit returns

```suggestion
    .map((workspaceMeta) => path.basename(workspaceMeta.location));
```